src/rules/flake8_logging_format/rules.rs

```diff
@@ -0,0 +1,238 @@
+    "relativeCreated",
+    "stack_info",
+    "thread",
+    "threadName",
+];
+
+/// Check logging messages for violations
+fn check_msg(checker: &mut Checker, msg: &Expr) {
+    match &msg.node {
+        // Check for string concatenation and percent format
+        ExprKind::BinOp { op, .. } => match op {
+            Operator::Add => {
+                if checker.settings.rules.enabled(&Rule::LoggingStringConcat) {
+                    checker.diagnostics.push(Diagnostic::new(
+                        LoggingStringConcat,
+                        Range::from_located(msg),
+                    ));
+                }
+            }
+            Operator::Mod => {
+                if checker.settings.rules.enabled(&Rule::LoggingPercentFormat) {
+                    checker.diagnostics.push(Diagnostic::new(
+                        LoggingPercentFormat,
+                        Range::from_located(msg),
+                    ));
+                }
+            }
+            _ => {}
+        },
+        // Check for f-strings
+        ExprKind::JoinedStr { .. } => {
+            if checker.settings.rules.enabled(&Rule::LoggingFString) {
+                checker
+                    .diagnostics
+                    .push(Diagnostic::new(LoggingFString, Range::from_located(msg)));
+            }
+        }
+        // Check for .format() calls
+        ExprKind::Call { func, .. } => {
+            if checker.settings.rules.enabled(&Rule::LoggingStringFormat) {
+                if let ExprKind::Attribute { value, attr, .. } = &func.node {
+                    if attr == "format" && matches!(value.node, ExprKind::Constant { .. }) {
+                        checker.diagnostics.push(Diagnostic::new(
+                            LoggingStringFormat,
+                            Range::from_located(msg),
+                        ));
+                    }
+                }
+            }
+        }
+        _ => {}
+    }
+}
+
+/// Check contents of the `extra` argument to logging calls
+fn check_log_record_attr_clash(checker: &mut Checker, extra: &Keyword) {
+    match &extra.node.value.node {
+        ExprKind::Dict { keys, .. } => {
+            for key in keys {
+                if let Some(key) = &key {
+                    if let ExprKind::Constant {
+                        value: Constant::Str(string),
+                        ..
+                    } = &key.node
+                    {
+                        if RESERVER_ATTRS.contains(&string.as_str()) {
+                            checker.diagnostics.push(Diagnostic::new(
+                                LoggingExtraAttrClash(string.to_string()),
+                                Range::from_located(key),
+                            ));
+                        }
+                    }
+                }
+            }
+        }
+        ExprKind::Call { func, keywords, .. } => {
+            if let ExprKind::Name { id, .. } = &func.node {
+                if id == "dict" {
```

Comment:
Let's use `checker.resolve_call_path(expr).map_or(false, |call_path| call_path.as_slice() == ["", "dict"])` here instead. That way, it'll be robust to overrides on `dict` (not common, but good to be consistent).

Reply:
Makes sense, thanks for the suggestion. Done: ad0e93bf5a2a075bedef3a8377a1378ff65a4ad2